### PR TITLE
Rework setup

### PR DESCRIPTION
**Pull request by mbakke**
* Move away from `setup.py` in favor of `setup.cfg`.
* Use `importlib.metadata` to set `__version__`.
* Add PEP517 compliance, so you can do `python -m build`.
* Fix CI.

### CODE REVIEW

File: zabbix_auto_config/__init__.py [hunk 1]
```diff
@@ -11,11 +11,13 @@
 import multiprocessing_logging
 import tomli
 
-from .__version__ import __version__
 from . import exceptions
 from . import models
 from . import processing
 
+import importlib.metadata
```
Comment (paalbra) [inline zabbix_auto_config/__init__.py:18]: I guess this import should be placed up with the other stdlib imports?

**Review by paalbra — COMMENTED** [review]
I have no experience with pyproject.toml etc. If it works it works? :)

Only a minor import comment from me

File: zabbix_auto_config/__init__.py (hunk 1 above)
Comment (mbakke) [reply zabbix_auto_config/__init__.py]: that makes sense; done :+1:

**Comment by mbakke** [comment]
Indeed, the `pyproject.toml` should work with any PEP517 compliant build frontend and distro tooling.

Now, rumor has it that setup.cfg will go away too eventually, so it may only be a stop-gap...

**Review by paalbra — APPROVED** [review]

**Pull request merged**